### PR TITLE
Feature/app 771 accident insurance result screens

### PR DESCRIPTION
**Pull request by StylianosGakis**
### Checklist

- [X] Functionality is covered by an integration test
- [X] Functionality is accessible in engineering mode

Available to preview in engineering mode through showkase > Components > Cross Sell result

### CODE REVIEW

File: app/src/main/AndroidManifest.xml
Comment (StylianosGakis) [inline app/src/main/AndroidManifest.xml:195]: I simply copied the standard configuration from the rest of the activities. I assume this is what we use everywhere unless there are some specific requirements for some screens.

File: app/src/main/java/com/hedvig/app/feature/crossselling/ui/CrossSellingResultActivity.kt
Comment (StylianosGakis) [inline app/src/main/java/com/hedvig/app/feature/crossselling/ui/CrossSellingResultActivity.kt:17]: Don't think the the case that the thing wouldn't exist inside the parcelable is even possible, please do comment if we want to handle this somewhat differently

File: app/src/main/java/com/hedvig/app/feature/crossselling/ui/CrossSellingResultScreen.kt
Comment (StylianosGakis) [inline app/src/main/java/com/hedvig/app/feature/crossselling/ui/CrossSellingResultScreen.kt:32]: Do we wait for string resources to exist before merge?

File: app/src/main/AndroidManifest.xml
Comment (hedvigoscar) [reply app/src/main/AndroidManifest.xml:195]: That's a sensible thing to do 😉

However, this is how we 'most recently' do this 🙂

```suggestion
        <activity
            android:name=".feature.crossselling.ui.CrossSellingResultActivity"
            android:windowSoftInputMode="adjustResize" />
```

File: app/src/main/java/com/hedvig/app/feature/crossselling/ui/CrossSellingResultActivity.kt
Comment (hedvigoscar) [reply app/src/main/java/com/hedvig/app/feature/crossselling/ui/CrossSellingResultActivity.kt:17]: I think it's fine to throw an error here, as there's a clear 'misuse' of the `Activity` if there is no `CROSS_SELLING_RESULT` on the `Intent`.

Something like this, perhaps:

```kotlin
val crossSellingResult = intent.getParcelableExtra<CrossSellingResult>(CROSS_SELLING_RESULT) ?: throw IllegalArgumentException("Programmer error: CROSS_SELLING_RESULT not provided to ${this.javaClass.name}")
```

File: app/src/main/java/com/hedvig/app/feature/crossselling/ui/CrossSellingResultActivity.kt
Comment (hedvigoscar) [inline app/src/main/java/com/hedvig/app/feature/crossselling/ui/CrossSellingResultActivity.kt:28]: I'm a big fan of the functional syntax for functions in kotlin that simply have one return statement:

```suggestion
        fun newInstance(context: Context, crossSellingResult: CrossSellingResult) = Intent(context, CrossSellingResultActivity::class.java).apply {
                putExtra(CROSS_SELLING_RESULT, crossSellingResult)
            }
```

File: app/src/main/java/com/hedvig/app/feature/crossselling/ui/CrossSellingResultActivity.kt
Comment (hedvigoscar) [inline app/src/main/java/com/hedvig/app/feature/crossselling/ui/CrossSellingResultActivity.kt:34]: Perhaps we can move this to its own file 🙂

File: app/src/main/java/com/hedvig/app/feature/crossselling/ui/CrossSellingResultScreen.kt
Comment (hedvigoscar) [reply app/src/main/java/com/hedvig/app/feature/crossselling/ui/CrossSellingResultScreen.kt:32]: Yes 🙂

File: app/src/main/java/com/hedvig/app/feature/crossselling/ui/CrossSellingResultScreen.kt
Comment (hedvigoscar) [inline app/src/main/java/com/hedvig/app/feature/crossselling/ui/CrossSellingResultScreen.kt:142]: I think this needs to be wired up to something 😉

File: app/src/main/java/com/hedvig/app/ui/compose/designsystem/LargeOutlinedButton.kt
Comment (hedvigoscar) [inline app/src/main/java/com/hedvig/app/ui/compose/designsystem/LargeOutlinedButton.kt:25]: What do you think about putting `.fillMaxWidth()` on this modifier? I think every use-case for the `Large Outlined Button` is actually `fillMaxWidth()`, this might save us some effort on the call-sites 🤔

File: app/src/main/java/com/hedvig/app/feature/crossselling/ui/CrossSellingResultScreen.kt
Comment (hedvigoscar) [inline app/src/main/java/com/hedvig/app/feature/crossselling/ui/CrossSellingResultScreen.kt:84]: Is this logic correct? Shouldn't it be `crossSellingResult.startingDay == LocalDate.now()`?

(also, we should inject a `Clock` here in order to not have a hard dependency on the system clock 😄)

File: app/src/main/java/com/hedvig/app/feature/crossselling/ui/CrossSellingResultScreen.kt
Comment (hedvigoscar) [inline app/src/main/java/com/hedvig/app/feature/crossselling/ui/CrossSellingResultScreen.kt:45]: Given that the only use-case for `InformationSection` is inside `CrossSellingResultScreen`, perhaps we could inline the modifier? Or is it hoisted because we lack the `BoxScope` inside `InformationSection`?

File: app/src/main/java/com/hedvig/app/feature/crossselling/ui/CrossSellingResultScreen.kt
Comment (hedvigoscar) [inline app/src/main/java/com/hedvig/app/feature/crossselling/ui/CrossSellingResultScreen.kt:161]: I'm generally a fan of not showing the system UI in previews as it takes up valuable screen real estate. What's your opinion on showing the system UI? 🙂

File: app/src/main/java/com/hedvig/app/feature/crossselling/ui/CrossSellingResultScreen.kt
Comment (hedvigoscar) [inline app/src/main/java/com/hedvig/app/feature/crossselling/ui/CrossSellingResultScreen.kt:43]: I wonder if this `Composable` is abstracting along the right axis.

I think it could be made a lot simpler by having a high-level check on `CrossSellResult`s variant, and then barely making any logic checks at later stages 🤔 

If you'd like, we can pair programming and I can show what I mean in practice 🙂

File: app/src/main/java/com/hedvig/app/feature/crossselling/ui/CrossSellingResultScreen.kt
Comment (hedvigoscar) [inline app/src/main/java/com/hedvig/app/feature/crossselling/ui/CrossSellingResultScreen.kt:101]: Perhaps moreso a comment to the designers, but I'm not sure that `18 dp` is a size we want to use in practice. There are some issues with translating the spacing in Figma to implementation that I'd like to show you as well, in particular with regards to the concept of `line height`, which does not work as Figma thinks it does in android 🙄

**Review by hedvigoscar — CHANGES_REQUESTED** [review]
Good-looking first PR! 🎉

File: app/src/main/java/com/hedvig/app/feature/crossselling/ui/CrossSellingResultScreen.kt
Comment (StylianosGakis) [reply app/src/main/java/com/hedvig/app/feature/crossselling/ui/CrossSellingResultScreen.kt:161]: In this particular case, not showing it looks like this
image
and showing it looks like this
image
That is because this screen is .fillMaxSize() anyway since it has a box filling the entire screen and the two different items are aligned to top and bottom respectively.

If anything might need changing here might be the CrossSellingResultScreen implementation if you have some ideas to do this better

Comment (hedvigoscar) [reply app/src/main/java/com/hedvig/app/feature/crossselling/ui/CrossSellingResultScreen.kt:161]: Oh, strange! Looks like the screen is less tall when system UI is shown? 😄 then the saving are moot I guess, this works totally fine 🙂

File: app/src/main/java/com/hedvig/app/feature/crossselling/ui/CrossSellingResultScreen.kt
Comment (StylianosGakis) [reply app/src/main/java/com/hedvig/app/feature/crossselling/ui/CrossSellingResultScreen.kt:43]: I think I see what you mean. Calculating what the two texts and the icon should be higher up and simply passing them down to InformationSection instead of passing the CrossSellResult and doing it all down there. We can look into it yes.

File: app/src/main/java/com/hedvig/app/feature/crossselling/ui/CrossSellingResultScreen.kt
Comment (StylianosGakis) [reply app/src/main/java/com/hedvig/app/feature/crossselling/ui/CrossSellingResultScreen.kt:101]: Yeah, I can show you how I got that number from Figma and you can tell me if I made some wrong assumptions on that process.

File: app/src/main/java/com/hedvig/app/feature/crossselling/ui/CrossSellingResultScreen.kt
Comment (StylianosGakis) [reply app/src/main/java/com/hedvig/app/feature/crossselling/ui/CrossSellingResultScreen.kt:45]: InformationSection could absolutely be an extension function to BoxScope which would allow this. This is in fact how I wrote it initially. The reason the modifier is hoisted is because then by looking at the Box() composable you can immediately know how they are arranged without looking at the two smaller composables themselves. In fact I'd say they themselves shouldn't really care how they are aligned in the box scope, and should just know how to render themselves, and the modifier coming from above can tell them their position in the box scope.
Does the thought process make sense? We can look at it together too and see the alternatives

File: app/src/main/AndroidManifest.xml
Comment (StylianosGakis) [reply app/src/main/AndroidManifest.xml:195]: How does it work without providing the theme to the activity? Don't we get problems with that?

Comment (hedvigoscar) [reply app/src/main/AndroidManifest.xml:195]: If we do not provide a theme on the `<activity />`-level, the `Activity` will use the `android:theme` defined for the `<application />` 🙂

File: app/src/main/java/com/hedvig/app/ui/compose/designsystem/LargeContainedButton.kt
Comment (hedvigoscar) [inline app/src/main/java/com/hedvig/app/ui/compose/designsystem/LargeContainedButton.kt:24]: Nice trick!

File: app/src/main/java/com/hedvig/app/feature/crossselling/ui/CrossSellingResultScreen.kt
Comment (StylianosGakis) [reply app/src/main/java/com/hedvig/app/feature/crossselling/ui/CrossSellingResultScreen.kt:155]: Absolutely, but this would be part of "wiring it up to the rest of the app" which I'd need some help on. Since right now there's no "normal" way to go into this screen anyway

**Pull request merged**
